frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPartition.tsx

```diff
@@ -0,0 +1,223 @@
+  value,
+  onChange,
+  onShowWidget,
+  getColumnTitle,
+  partitions,
+  columns,
+}: {
+  value: ColumnNameColumnSplitSetting;
+  onChange: (value: ColumnNameColumnSplitSetting) => void;
+  onShowWidget: (
+    widget: { id: string; props: { initialKey: string } },
+    ref: HTMLElement | undefined,
+  ) => void;
+  getColumnTitle: (column: DatasetColumn) => string;
+  columns: RemappingHydratedDatasetColumn[];
+  partitions: Partition[];
+}) => {
+  const theme = useMantineTheme();
+
+  const handleEditFormatting = (
+    column: RemappingHydratedDatasetColumn,
+    targetElement: HTMLElement,
+  ) => {
+    if (column) {
+      onShowWidget(
+        {
+          id: "column_settings",
+          props: {
+            initialKey: getColumnKey(column),
+          },
+        },
+        targetElement,
+      );
+    }
+  };
+
+  const getPartitionType = (
+    partitionName: keyof ColumnNameColumnSplitSetting,
+  ) => {
+    switch (partitionName) {
+      case "rows":
+      case "columns":
+        return "dimension";
+      default:
+        return "metric";
+    }
+  };
+
+  const handleDragEnd: OnDragEndResponder = ({ source, destination }) => {
+    if (!source || !destination) {
+      return;
+    }
+    const { droppableId: sourcePartition, index: sourceIndex } = source;
+    const { droppableId: destinationPartition, index: destinationIndex } =
+      destination;
+
+    if (
+      sourcePartition === destinationPartition &&
+      sourceIndex !== destinationIndex
+    ) {
+      onChange({
+        ...value,
+        [sourcePartition]: columnMove(
+          value[sourcePartition as keyof ColumnNameColumnSplitSetting],
+          sourceIndex,
+          destinationIndex,
+        ),
+      });
+    } else if (sourcePartition !== destinationPartition) {
+      const column =
+        value[sourcePartition as keyof ColumnNameColumnSplitSetting][
+          sourceIndex
+        ];
+
+      onChange({
+        ...value,
+        [sourcePartition]: columnRemove(
+          value[sourcePartition as keyof ColumnNameColumnSplitSetting],
+          sourceIndex,
+        ),
+        [destinationPartition]: columnAdd(
+          value[destinationPartition as keyof ColumnNameColumnSplitSetting],
+          destinationIndex,
+          column,
+        ),
+      });
+    }
+  };
+
+  const updatedValue = useMemo(
+    () =>
+      _.mapObject(value || {}, columnNames =>
+        columnNames
+          .map(columnName => columns.find(col => col.name === columnName))
+          .filter((col): col is RemappingHydratedDatasetColumn => col != null),
+      ),
+    [columns, value],
+  );
+
+  return (
+    <DragDropContext onDragEnd={handleDragEnd}>
+      {partitions.map(({ name: partitionName, title }, index) => {
+        const updatedColumns = updatedValue[partitionName] ?? [];
+        const partitionType = getPartitionType(partitionName);
+        return (
+          <Box
+            className={cx(CS.py2, { [CS.borderTop]: index > 0 })}
```

Comment:
Would `py="md"` work instead of `CS.py2`?

Reply:
yes!